### DISCUSSION TITLE
Server express middlewares is not working

### DISCUSSION THREAD

**Discussion by kibo-sushant** [nuxt/nuxt#18350] [Questions]
The existing code is in Nuxt 2, I am running express along with Nuxt and its working fine.
I migrated code on Nuxt3 using Bridge (https://v3.nuxtjs.org/getting-started/bridge) after that it stopped working.

**Steps followed (to configure express: https://blowstack.com/blog/how-to-integrate-express_js-with-nuxt-app)**
1. _npm install express --save_
2. _api/index.js_

const express = require('express')

const app = express()
app.use(express.json())
app.use(express.urlencoded({ extended: false }))

app.get('/test', function (req, res) {
  res.send('Test successful')
})

export default {  path: '/api',  handler: app }

3. _nuxt.config.js_
...
/*  Server Middleware */
serverMiddleware: [{ path: "/", handler: "~/api" }]
...

**Issue**
 After migrating when I remove serverMiddleware above code, it works fine 
But with the above serverMiddleware: code, abrowser just shows loading

**Comment by lucpotage** [comment]
Server middleware should go under `server/middleware`. `server/api` is reserved for API endpoints.
https://v3.nuxtjs.org/docs/directory-structure/server#server-middleware

Does it help?

**Comment by tobiasdiez** [comment]
I have exactly the same issue at https://github.com/JabRef/JabRefOnline/pull/480: the browser shows an infinite loading screen.
However, I failed to create a minimal reproducible example so far. The following code sandbox https://codesandbox.io/s/serene-tree-ofwn9 works just fine (essentially using your code), with the client running at https://ofwn9-43013.sse.codesandbox.io/ and the api at https://ofwn9-43013.sse.codesandbox.io/api/test.

Feel free to fork this sandbox.
